src/centos/7/source-build/Dockerfile

```diff
@@ -0,0 +1,60 @@
+FROM centos:7
+
+# Install dependencies
+
+RUN yum install -y \
+        centos-release-scl \
+        epel-release \
+        scl-utils-build \
+    && \
+    yum install -y \
+        "perl(Time::HiRes)" \
+        autoconf \
+        automake \
+        devtoolset-7-gcc* \
+        doxygen \
+        gdb \
+        git \
+        krb5-devel \
+        libcurl-devel \
+        libedit-devel \
+        libgdiplus \
+        libicu-devel \
+        libidn-devel \
+        libmetalink-devel \
+        libnghttp2-devel \
+        libssh2-devel \
+        libtool \
+        libunwind-devel \
+        libuuid-devel \
+        libxml2-devel \
+        lttng-ust-devel \
+        lzma \
+        make \
+        ncurses-devel \
+        numactl-devel \
+        openssl-devel \
+        python-argparse \
+        python-devel \
+        readline-devel \
+        python3 \
+        sudo \
+        swig \
+        wget \
+        which \
+        xz \
+        zlib-devel \
+        clang \
+        llvm \
+    && \
+    yum clean all
+
+# Add cmake 3.6.2. This is not including in the package manager, yum install 
+# cmake will include 2.8
+
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.6.2/cmake-3.6.2-Linux-x86_64.tar.gz && \
+    tar -xvf cmake-3.6.2-Linux-x86_64.tar.gz && \
+    mv cmake-3.6.2-Linux-x86_64 /cmake
+
```

Comment:
IIRC there's a `cmake3` package that comes as part of the distribution. We should use that if possible. That way, we're using only packages available in the distro.

cc: @omajid

Reply:
Will see what the version is. Checking now

Reply:
I don't see `cmake3` here: http://mirror.centos.org/centos/7/os/x86_64/Packages/. I do see `cmake3` in EPEL (extra packages for CentOS/RHEL): https://dl.fedoraproject.org/pub/epel/7/x86_64/Packages/c/, but that's 3.17.

Reply:
Right we found this is 3.17, we will continue with the download approach to get 3.6.2.

Reply:
There's a `llvm-toolset-7-cmake` (version 3.6.2) package that's available, but using that `cmake` is a bit harder and error-prone. You have to sprinkle `scl enable ... ` commands liberally in your builds. Btw, this is also the package we use when building internally at Red Hat.

I suggest going with the download approach for now.